### PR TITLE
More intuitive environment variable handling

### LINKED ISSUE #170: DB initializing only checks whether environment variable is set, but not whether it is 1 or 0
**Describe the bug**

When messing around with the setup I set the FLASK_DEBUG env to 0. 
However, MegaQC expects it to be unset. 

**Expected behavior**

I expect MegaQC to treat those special ENVs as unset if equal to 0.

### PR DESCRIPTION
**Pull request by Zethson**
Hi,

fixes #170 .

```
export FLASK_DEBUG=0                                                                                                           
export MEGAQC_PRODUCTION=1                                                                                                  

>>> from environs import Env
>>> env = Env()
>>> env.read_env()
>>> print(env.bool("FLASK_DEBUG", False))
False
>>> print(env.bool("MEGAQC_PRODUCTION", False))
True
```

Signed-off-by: Zethson <lukas.heumos@posteo.net>

### CODE REVIEW

File: megaqc/cli.py [hunk 1]
```diff
@@ -38,10 +42,14 @@ def cli():
...
-    if os.environ.get("FLASK_DEBUG", False):
+
+    env = Env()
+    env.read_env()
+
+    if env.bool("FLASK_DEBUG", False):
         print(" * Environment variable FLASK_DEBUG is true - running in dev mode")
         os.environ["FLASK_ENV"] = "dev"
-    elif not os.environ.get("MEGAQC_PRODUCTION", False):
+    elif env.bool("MEGAQC_PRODUCTION", False):
```
Comment (TMiguelT) [inline megaqc/cli.py:52]: Missing the `not`

**Review by TMiguelT — CHANGES_REQUESTED** [review]
You will also need to look into `environs`-ifying other uses of `os.environ`, for consistent behaviour:
* `megaqc/wsgi.py`
* `megaqc/settings.py`
* `megaqc/utils/settings.py`

Also, the docs will need to explain somewhere about the `.env` file.

**Comment by Zethson** [comment]
@TMiguelT Thanks for the review.

I've added consistent env reading to the other files and hope that I didn't miss any.
Tested it briefly and it seems to do what it is supposed to do.

Honestly don't think that we need the `.env` file reading here and therefore I removed it.

File: megaqc/settings.py [hunk 2]
```diff
@@ -109,11 +112,11 @@ class ProdConfig(Config):
         SQLALCHEMY_HOST = os.environ["DB_UNIX_SOCKET"]
     else:
         SQLALCHEMY_HOST = "{}:{}".format(
-            os.environ.get("DB_HOST", "localhost"), os.environ.get("DB_PORT", "5432")
+            env.str("DB_HOST", "localhost"), env.str("DB_PORT", "5432")
```
Comment (TMiguelT) [inline megaqc/settings.py:115]: Should the port be an `int` in order to get a better error message?

**Review by TMiguelT — APPROVED** [review]

**Pull request merged**